src/scrape/scrape_job_test.go

```diff
@@ -144,18 +145,50 @@ var kubeletSpecs = definition.SpecGroups{
 	"container": metric.KubeletSpecs["container"],
 }
 
-type testGrouper struct{}
+type testGrouper struct {
+	// PodValuesInTime allows to overwrite some metrics value in each Group execution. Ex: {"isReady": {0, 1, 2, 3}}
+	PodValuesInTime map[string][]interface{}
+	// ContainerValuesInTime allows to overwrite some metrics value in each Group execution. Ex: {"restartCount": {0, 1, 2, 3}}
+	ContainerValuesInTime map[string][]interface{}
```

Comment:
Is there a reason to have two identical fields for pod and container specifically? I wonder if we could do this differently so if we need to add values over time for another entity, e.g. `Volume`, we don't need to add another `VolumeValuesInTime` field.

Reply:
After this change [50c8a5a](https://github.com/newrelic/nri-kubernetes/pull/382/commits/50c8a5a2a9cebd84bf71f7861bc835b414a5a448) it is a bit more abstract.

We could go further and extract this mock and the corresponding specs to a _testutils file_ and give them some parameters to make it easier to reduce fixtures as needed in different tests... I would include that change in a different PR.